Source/WebCore/page/LocalFrame.cpp

```diff
@@ -1599,6 +1599,21 @@ RefPtr<SecurityOrigin> LocalFrame::frameDocumentSecurityOrigin() const
     return nullptr;
 }
 
+Vector<Ref<HistoryItem>> LocalFrame::backForwardList() const
+{
+    return m_backForwardList;
+}
+
+void LocalFrame::setBackForwardList(Vector<Ref<HistoryItem>>&& list)
+{
+    m_backForwardList = WTFMove(list);
+}
+
+void LocalFrame::addItemToBackForwardList(Ref<HistoryItem>&& item)
+{
+    m_backForwardList.append(WTFMove(item));
```

Comment:
Is this correct? I think there are some cases where you need to add an item to the middle of a back-forward list and then toss everything from that point in the list moving forward. (See `WebBackForwardList::addItem`).